R/select.R

```diff
@@ -0,0 +1,155 @@
+#' Select variables using `select`
+#'
+#' `step_select` creates a *specification* of a recipe step
+#'  that will select variables using [dplyr::select()].
+#'
+#' @inheritParams step_center
+#' @param ... One or more selector functions to choose which
+#'  variables that will be evaluated by the filtering bake. See
+#'  [selections()] for more details. For the `tidy`
+#'  method, these are not currently used.
+#' @param role For model terms selected by this step, what analysis
+#'  role should they be assigned?
+#' @param inputs Quosure(s) of `...`.
+#' @return An updated version of `recipe` with the new step
+#'  added to the sequence of existing steps (if any). For the
+#'  `tidy` method, a tibble with column `terms` which
+#'  contains the `select` expressions as character strings
+#'  (and are not reparsable).
+#' @details When an object in the user's global environment is
+#'  referenced in the expression defining the new variable(s),
+#'  it is a good idea to use quasiquotation (e.g. `!!`) to embed
+#'  the value of the object in the expression (to be portable
+#'  between sessions). See the examples.
+#' @keywords datagen
+#' @concept preprocessing
+#' @concept variable_filters
+#' @export
+#' @examples
+#' rec <-
+#'   recipe(~., data = iris) %>%
+#'   step_select(Species, starts_with("Sepal"))
+#'
+#' prepped <- prep(rec, training = iris %>% slice(1:75))
+#'
+#' library(dplyr)
+#'
+#' dplyr_train <-
+#'   iris %>%
+#'   as_tibble() %>%
+#'   slice(1:75) %>%
+#'   select(Species, starts_with("Sepal"))
+#'
+#' rec_train <- bake(prepped, new_data = NULL)
+#' all.equal(dplyr_train, rec_train)
+#'
+#' dplyr_test <-
+#'   iris %>%
+#'   as_tibble() %>%
+#'   slice(76:150) %>%
+#'   select(Species, starts_with("Sepal"))
+#' rec_test <- bake(prepped, iris %>% slice(76:150))
+#' all.equal(dplyr_test, rec_test)
```

Comment:
I think this can be a little more compactly written as:

```r
library(dplyr)

iris_tbl <- as_tibble(iris)
iris_train <- slice(iris_tbl, 1:75)
iris_test <- slice(iris_tbl, 76:150)

dplyr_train <- select(iris_train, Species, starts_with("Sepal"))
dplyr_test <- select(iris_test, Species, starts_with("Sepal"))

rec <- recipe(~., data = iris_train) %>%
  step_select(Species, starts_with("Sepal")) %>%
  prep(training = iris_train)

rec_train <- bake(rec, new_data = NULL)
all.equal(dplyr_train, rec_train)

rec_test <- bake(rec, iris_test)
all.equal(dplyr_test, rec_test)
```

Reply:
Nice, thanks!

Reply:
I've used this pattern also in the tests now